src/hw_isolation_record/manager.cpp

```diff
@@ -553,38 +558,34 @@ void Manager::handleHostIsolatedHardwares()
         return;
     }
 
-    std::for_each(records.begin(), records.end(), [this](const auto& record) {
-        // Make sure the isolated hardware record is resolved (recordId will
-        // get change as "0xFFFFFFFF") by host OR the record is already exist.
-        auto isolatedHwIt = std::find_if(
-            _isolatedHardwares.begin(), _isolatedHardwares.end(),
-            [&record](const auto& isolatedHw) {
-                return (
-                    (isolatedHw.second->getEntityPath() == record.targetId) &&
-                    ((record.recordId == 0xFFFFFFFF) ||
-                     (isolatedHw.second->getEntryRecId() == record.recordId)));
-            });
-
-        // If not found in the existing isolated hardware entries list then
-        // the host created a new record for isolating some hardware so
-        // add the new dbus entry for the same.
-        if (isolatedHwIt == _isolatedHardwares.end())
-        {
-            this->createEntryForRecord(record);
-        }
-        else if (record.recordId == 0xFFFFFFFF)
-        {
-            // Update Resolved and Enabled properties respectively
-            // in the hardware isolation entry and inventory D-Bus objects
-            isolatedHwIt->second->resolveEntry(false);
-        }
-        else
+    /**
+     * First sync all D-Bus entries with retrieved records because
+     * the existing record might be overwritten during creation if that's meets
+     * certain overwritten conditions.
+     */
+    IsolatedHardwares::iterator entryIt = _isolatedHardwares.begin();
+    openpower_guard::GuardRecords::iterator recordIt = records.begin();
+    for (; entryIt != _isolatedHardwares.end(); entryIt++, recordIt++)
```

Comment:
is there a 1 to 1 mapping between these two lists?

Reply:
Yes, because we are not allowed to delete a record, just invalidating (aka resolving through Record id i.e changing it as 0xFFFFFFFF).

i.e if the entry is created for the new record it stays forever until reset.

FYI, if the file is fulled and there is resolved record slot, that used to overwrite the new record

- No records
- Updated the existing record slot.
- New record.